### PR TITLE
Move from Travis to GitHub Actions CI

### PR DESCRIPTION
**Pull request by auvipy**

### CODE REVIEW

File: .github/workflows/python-publish.yml
Comment (cclauss) [inline .github/workflows/python-publish.yml:26]:
```suggestion
      uses: actions/setup-python@v4
```

**Comment by auvipy** [comment]
IMHO, link and test actions should be different

**Comment by cclauss** [comment]
Sorry.  I proposed the `setuptools` upgrade in the wrong file... #854

**Comment by cclauss** [comment]
Please rebase because `lint_python` should now pass.

File: .github/workflows/python-build.yml
Comment (cclauss) [inline .github/workflows/python-build.yml:32]: The name does not seem to be aligned with the command.  It is self-documenting to just remove the name.
```suggestion
      - run: sudo apt install -y graphviz
```

**Comment by JonathanHuot** [comment]
Looks good to me, next step would be to validate tag/publish once the release is ready. In the meantime, once reviewed, I propose to merge the PR and integrate other pending PRs (rebase) for release preparation.

**Comment by auvipy** [comment]
Jonathon, this needs yours approval for merge

**Comment by cclauss** [comment]
# This branch has conflicts that must be resolved

Please rebase on the current `master`

**Comment by auvipy** [comment]
I should not merge the ruff one first!

**Comment by cclauss** [comment]
You might try manually undoing all changes to `.github/workflows/lint_python.yml` (replace all the green lines with the red lines) and that might cure the git conflict.  Or update your `master` branch and then:
```
git checkout CI
git pull  # just to be sure
git checkout master -- .github/workflows/lint_python.yml
git commit -am"Update .github/workflows/lint_python.yml"
```

**Review by JonathanHuot — APPROVED** [review]
LGTM CI passing with success

**Pull request merged**
